From 5e9053988ff76b0eb2c2ffba4214189899df3c65 Mon Sep 17 00:00:00 2001
From: Ben Brown <ben.brown@codethink.co.uk>
Date: Mon, 24 Aug 2020 18:44:03 +0100
Subject: [PATCH] Fix module entry point declaration

Explicit definition of a py::module was only required when using the
deprecated PYBIND11_PLUGIN macro; remove this as the second argument of
the PYBIND11_MODULE macro will define the variable.

Set the docstring as appropriate.

Fixes #404.
---
 api/python/pyLIEF.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/api/python/pyLIEF.cpp b/api/python/pyLIEF.cpp
index 0ade098..5978d8c 100644
--- a/api/python/pyLIEF.cpp
+++ b/api/python/pyLIEF.cpp
@@ -50,13 +50,12 @@
 #include "platforms/android/pyAndroid.hpp"
 
 
-py::module LIEF_module("lief", "Python API for LIEF");
-
 PYBIND11_MODULE(lief, LIEF_module) {
 
   LIEF_module.attr("__version__")   = py::str(LIEF_VERSION);
   LIEF_module.attr("__tag__")       = py::str(LIEF_TAG);
   LIEF_module.attr("__is_tagged__") = py::bool_(LIEF_TAGGED);
+  LIEF_module.doc() = "Python API for LIEF";
 
   init_LIEF_Object_class(LIEF_module);
 
-- 
2.28.0

